apps/analysis_framework/serializers.py

```diff
@@ -126,15 +130,82 @@ def validate_project(self, project):
 
     def create(self, validated_data):
         project = validated_data.pop('project', None)
+        private = validated_data.get('is_private', False)
+
+        # Check if user has access to private project feature
+        user = self.context['request'].user
+        private_access = user.profile.get_accessible_features().filter(
+            key=Feature.PRIVATE_PROJECT
+        ).exists()
+
+        if private and not private_access:
+            raise exceptions.PermissionDenied({
+                'message': 'You can not create private frameworks.'
```

Comment:
Use similar message format (i.e. to that of private project)

Reply:
Fixed.